### ISSUE TITLE
optional not working with nested case classes?

### ISSUE THREAD

**Issue by nornagon** [kailuowang/henkan#43]
The following gives a compiler error:

```scala
import $ivy.`com.kailuowang::henkan-convert:0.6.1`
import $ivy.`com.kailuowang::henkan-optional:0.6.1`

import henkan.optional.all._
import cats.data.Validated
import cats.implicits._

case class Foo(bar: String)
case class Message(a: Option[Foo], b: Option[Int])
case class Domain(a: Foo, b: Int)

validate(Message(Some(Foo("")), Some(2))).to[Domain]
```

```
test.sc:12: ambiguous implicit values:
 both method mkGenValidateFromOptional in trait MkValidateFromOptional of type [From, To, FL <: shapeless.HList, TL <: shapeless.HList](implicit genFrom: shapeless.LabelledGeneric.Aux[From,FL], implicit genTo: shapeless.LabelledGeneric.Aux[To,TL], implicit convertHList: shapeless.Lazy[henkan.optional.ValidateFromOptional[FL,TL]])henkan.optional.ValidateFromOptional[From,To]
 and method mkFromIdentity in trait MkValidateFromOptional1 of type [V]=> henkan.optional.ValidateFromOptional[V,V]
 match expected type henkan.optional.ValidateFromOptional[ammonite.$file.test.Foo,ammonite.$file.test.Foo]
val res_8 = validate(Message(Some(Foo("")), Some(2))).to[Domain]
    ^
test.sc:12: Cannot build validate function from ammonite.$file.test.Message to ammonite.$file.test.Domain, possibly due to missing fields in ammonite.$file.test.Message or missing cats instances (`Traverse` instances are needed to convert fields in containers)
val res_8 = validate(Message(Some(Foo("")), Some(2))).to[Domain]
                                                        ^
```

Replacing `Foo` by `String` (as in the examples) works fine though. Is there something special I need to do to have henkan convert nested case classes with optionals?

**Comment by kailuowang** [comment]
Thanks very much for reporting with the minimized example.  Let me investigate and get back to you.

**Comment by kailuowang** [comment]
Will release soon

**Comment by kailuowang** [comment]
released 0.6.2 which should fix this.

**Comment by nornagon** [comment]
Whoa, that was quick! Thanks!
